### ISSUE TITLE
Reproducing Table 4 of supplementary material

### ISSUE THREAD

**Issue by slippylolo** [krrish94/nerf-pytorch#11]
Hey, 

Thank you for the awesome work on this implementation.

I have been playing around with it quite a bit. I however can't get any results close to that of Table 4, found in the supplementary material of the paper. For instance, they report a PSNR of ~30 on the Lego scene of their synthetic dataset. 

In my experiments, either using the pre-trained model of yours or my own trained from scratch, I get a PSNR of only ~20.

**Comment by krrish94** [comment]
@slippylolo For the examples the authors run in the paper, the configs are quite different (and huge!)

Maybe this could help? https://github.com/bmild/nerf/tree/master/paper_configs

**Comment by slippylolo** [comment]
Huge is no problem, I have some spare V100s :) 

I have checked this config and reproduced it (to the best of my knowledge) with little success. Adding 128 fine samples doesn't help much, and training to 500k doesn't change much either. I have also tried with the full resolution images, and it doesn't help dramatically either. 

From his readme, https://github.com/kwea123/nerf_pl has apparently succeeded. If the config presented in the readme is indeed the one with ~30 PSNR, it doesn't look like much additional compute. 

If I can get a model to ~30 PSNR, I will contribute it to the repo--it's a bit of shame that the SOTA for NeRF is not easily available right now.

**Comment by krrish94** [comment]
@slippylolo That'd be fantastic!

I'm currently pressed for time and compute, and plan on working on this repo in 3-4 weeks from now. It'll be sweet if you're able to repro the ablation results. Also, this might be relevant: https://github.com/bmild/nerf/issues/29#issuecomment-614779498

**Comment by slippylolo** [comment]
Thank you for pointing this issue to me!

I believe I have found the culprit behind the PSNR not matching what was reported in the paper. 

Indeed, at validation, the PSNR reported in your code is the combination of the coarse + fine losses. Obviously this value is much lower than the actual PSNR, since it's roughly doubling the loss!

The fine model loss is in fact more in line with a PSNR of ~25 (both on your pretrained model and my own trained from scratch). While that's not the ~30 of the paper, that's much closer--and the extra iterations they have probably bridge that gap. 

Thank you for the prompt answers to my issue. I will be closing it and contribute in the coming days/weeks some improvements and pretrained models I have! (I need to clean-up some stuff first and also have a deadline to meet :)).

**Comment by jeromepl** [comment]
@slippylolo any update on reproducing the ~30 PSNR of the paper? I have been training a model up to 500k iterations and am still hovering around 28 validation PSNR (for the fine model).
I have been training (and validating) on 800x800 px images, with N_importance=128 to reproduce the NeRF paper but this other implementation you mentioned which achieved ~30 PSNR ran with 400x400 px and N_importance=64. Do you think this could explain the difference?

**Comment by slippylolo** [comment]
Hey @jeromepl, 
Sadly I haven't succeeded in reproducing the ~30 PSNR. In my experience, the N_importance and the resolution of the image didn't matter much. I haven't done much with NeRF in the past month or so, but I ended up thinking that there might be something wrong in this implementation explaining the performance gap -- but I haven't had time to do any proper debugging, so it's just a hunch.

**Comment by kwea123** [comment]
As I mentioned in #6, the config is not fully applied and the network size in this implementation is much smaller than the paper version, which explains its high speed and low performance... I suspect that he has not corrected at all.

**Comment by slippylolo** [comment]
Yes, I noticed that as well, but fixed it on my side. I however agree it’s quite possible something else is missing/inappropriately applied somewhere.
